circuits/src/mpc_gadgets/comparators.rs

```diff
@@ -15,28 +15,33 @@ use super::{
 
 /// Implements the comparator a < 0
 ///
-/// D represents is the bitlength of the input
+/// D represents is the bitlength of the input values
+///
+/// Note that this definition is only correct if we allow for positive values
+/// only up to 252 bits. This is because we define a scalar as negative if it
+/// is greater than (p-1)/2, then for our 254 bit prime, this threshold sits
+/// between values 2^252 and 2^253. Therefore, our method of truncation and
+/// top-two bit check will fail for values 2^252 <= x < (p-1) / 2
+///
+/// So we enforce that the values are less than 2^252
 pub fn less_than_zero<const D: usize>(
-    a: &AuthenticatedScalarResult,
-    fabric: &MpcFabric,
-) -> AuthenticatedScalarResult {
+    a: &AuthenticatedScalar,
+    fabric: &Fabric,
+) -> AuthenticatedScalar {
+    assert!(D <= SCALAR_MAX_BITS - 2, "D mus");
     // Truncate the first 250 bits of the input
-    let truncated = truncate::<250>(a, fabric);
+    let truncated = truncate::<252>(a, fabric);
```

Comment:
nit: we should constantize this, and use the same constant in the assertion above